src/backend/pipeline/tuplebuf.c

```diff
@@ -614,7 +614,7 @@ TupleBufferTryWait(TupleBufferReader *reader)
 	SpinLockRelease(&buf->mutex);
 
 	if (should_wait)
-		WaitLatch(reader->proc->latch, WL_LATCH_SET | WL_POSTMASTER_DEATH, 0);
+		WaitLatch(reader->proc->latch, WL_LATCH_SET | WL_POSTMASTER_DEATH, 1);
```

Comment:
We'll they'll sleep for at least 1ms, so when they're not busy they'll spend the majority of their time asleep right?